components/ordhook-core/src/core/meta_protocols/brc20/index.rs

```diff
@@ -15,10 +15,66 @@ use super::{
     brc20_activation_height,
     cache::Brc20MemoryCache,
     parser::ParsedBrc20Operation,
-    verifier::{verify_brc20_operation, verify_brc20_transfer, VerifiedBrc20Operation},
+    verifier::{verify_brc20_operation, verify_brc20_transfers, VerifiedBrc20Operation},
 };
 
-/// Indexes BRC-20 operations in a Bitcoin block. Also writes the indexed data to DB.
+/// Index ordinal transfers in a single Bitcoin block looking for BRC-20 transfers.
+async fn index_unverified_brc20_transfers(
+    transfers: &Vec<(&TransactionIdentifier, &OrdinalInscriptionTransferData)>,
+    block_identifier: &BlockIdentifier,
+    timestamp: u32,
+    brc20_cache: &mut Brc20MemoryCache,
+    brc20_db_tx: &Transaction<'_>,
+    ctx: &Context,
+) -> Result<Vec<(usize, Brc20Operation)>, String> {
+    if transfers.is_empty() {
+        return Ok(vec![]);
+    }
+    let mut results = vec![];
```

Comment:
Also from a performance perspective, it takes less time if the capacity of the `vec` is preallocated, if its capacity is known.
```suggestion
    let mut results = Vec::with_capacity(transfers.len());
```

Reply:
In this case we don't know the final capacity because not all incoming transfers may be brc20 transfers, so the result Vec might even be empty at the end

Reply:
Got it.